gnomad/utils/filtering.py

```diff
@@ -257,20 +261,25 @@ def subset_samples_and_variants(
             f"IDs that aren't in the MT: {missing_samples}\n"
         )
 
-    mt = mt.semi_join_cols(sample_ht)
-    if sparse:
-        mt = mt.filter_rows(
-            hl.agg.any(mt[gt_expr].is_non_ref() | hl.is_defined(mt.END))
-        )
+    if isinstance(t, hl.vds.VariantDataset):
+        t = hl.vds.filter_samples(t, sample_ht, keep=True)
```

Comment:
I wonder if we should add an option for `remove_dead_alleles` that can only be used with a VDS?

Reply:
I think that would make sense, and set the default to True?

Reply:
Actually, because of the computation to do this it might actually be better to set it to False. If we don't care then no reason to do the extra computations. I think in most cases we will not care, for release of a subset with samples in the full callset that can't be released, this will need to be set to True